src/System.Net.Sockets/tests/FunctionalTests/NetworkStreamTest.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Sockets.Tests
+{
+    public class NetworkStreamTest
+    {
+        [Theory]
+        [MemberData("NonCanceledTokens")]
+        public async Task ReadWriteAsync_NonCanceled_Success(CancellationToken nonCanceledToken)
+        {
+            await RunWithConnectedNetworkStreamsAsync(async (server, client) =>
+            {
+                var clientData = new byte[] { 42 };
+                await client.WriteAsync(clientData, 0, clientData.Length, nonCanceledToken);
+
+                var serverData = new byte[clientData.Length];
+                Assert.Equal(serverData.Length, await server.ReadAsync(serverData, 0, serverData.Length, nonCanceledToken));
+
+                Assert.Equal(clientData, serverData);
+            });
+        }
+
+        [Fact]
+        public async Task ReadWriteAsync_Canceled_ThrowsOperationCanceledException()
```

Comment:
It's documented to throw ```OperationCanceledException```, and a proper implementation is likely to throw the concrete ```OperationCanceledException```, at least in some cases.  I just so happens that in the precanceled case, because of the specifics of the implementation right now, it will be a ```TaskCanceledException```, but I don't think the test should depend on that.